src/Ships/RepairCorvette.c

```diff
@@ -221,7 +221,7 @@ void stopRepairEffect(Ship *ship)
 
     if (bitTest(((etgeffectstatic *)ship->rceffect->staticinfo)->specialOps, ESO_SelfDeleting))
     {                                                       //if the effect will delete itself
-        ((real32 *)ship->rceffect->variable)[ETG_ResourceDurationParam] =
+        ((real32 *)ship->rceffect->variable)[SCALESHIFT(ETG_ResourceDurationParam)] =
```

Comment:
`SCALECAST` I have seen it used in relation to 64b code. However, I have no clue what it's accomplishing here.

Reply:
Oh wait I think I get it. real32 is still 4 bytes of size, but we're passing 8 bytes arguments to ETG functions. So when we're writing in the parameter array, we have to jump twice the size of a real32 before getting to the beginning of the next parameter.

Reply:
But then you should also update the ETG_ResourceLengthParam and ETG_ResourceRadiusParam (the positions of the other params)

Actually, why not apply the scale shift at their definition ?

Reply:
I have to confess that I didn't really know _why_ this fixed the issue. I'll check it again using you insights tomorrow.